_rules/aria-required-context-role-ff89c9.md

```diff
@@ -0,0 +1,284 @@
+
+```html
+<div role="list">
+    <div aria-label="menu">
+         <div role="listitem">List item 1</div>
+    </div>
+</div>
+```
+
+**Note:** This test case follows the definition of [owned by](#owned-by) used in this rule. If implemented differently, this definition could cause differences in outcome for this test case. 
+
+
+#### Failed Example 5
+
+Element with role `listitem` has a closer ancestor, that is included in the accessibility tree, than the role `list` that should have been its context role.
+
+```html
+<div role="list">
+  <div role="tabpanel">
+    <div role="listitem">List item 1</div>
+   </div>
+</div>
+```
+
+#### Failed Example 6
+
+The element with the semantic role of `listitem` is [owned by](#owned-by) the first element that references it through `aria-owns`, which results in the wrong context role.
+
+```html
+<div role="tabpanel" aria-owns="id1">
+  <div role="list" aria-owns="id1">
+    <div id="id1" role="listitem">List item 1</div>
+   </div>
+</div>
+```
+
+**Note:** This test case follows the definition of [owned by](#owned-by) used in this rule. If implemented differently, this definition could cause differences in outcome for this test case.
+
+#### Failed Example 7
+
+Since explicit ownership cannot cross shadow boundaries, the element with the semantic role of `listitem` does not have a context role.
+
+```html
+<div role="list" aria-owns="item">
+</div>
+
+<div id="host"></div>
+
+<script>
+  const host = document.querySelector("#host");
+  const root = host.attachShadow({ mode: "open" });
+
+  root.innerHTML = `
+    <div id="item" role="listitem">List item 1</div>
+  `;
+</script>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+Element does not have an explicit semantic role.
+
+```html
+<ul>List item 1</ul>
+```
+
+#### Inapplicable example 2
+
+Element is not exposed to assistive technologies.
+
+```html
+<div role="listitem" aria-hidden="true">List item 1</div>
+```
+
+#### Inapplicable Example 3
+
+Role does not have any required context roles listed in WAI-ARIA spec.
+
+```html
+<div role="radio">Radio button 1</div>
+```
+
+#### Inapplicable Example 4
+
+Element is not exposed to assistive technologies, since ancestor has `aria-hidden` attribute with value set to `true`.
+
+```html
+<div role="list" aria-hidden="true">
+    <div role="listitem">List item 1</div>
+</div>
+```
+#### Inapplicable Example 5
+
+Element has an explicit semantic role, but it is identical to the implicit semantic role, making the element inapplicable.
+
+```html
+<li role="listitem">List item 1</li>
+```
+
+#### Inapplicable Example 6
+
+Element has a role from the [Digital Publishing WAI-ARIA Module 1.0](https://www.w3.org/TR/dpub-aria-1.0/), not the [WAI-ARIA 1.1 Recommendation](https://www.w3.org/TR/wai-aria-1.1/), and it is therefore inapplicable.
+
+```html
+<div role="doc-bibliography">Item 1<div>
```

Comment:
Maybe include example 6 from DPUB-ARIA here? `doc-bibliography` doesn't have a required context role, so that's not a particularly meaningful example from DPUB. `doc-biblioentry` does though, so that might be worth including.

Reply:
Fixed